### ISSUE TITLE
RFuture.cause() method should return CancellationException and not throw it.

### ISSUE THREAD

**Issue by mrniko** [redisson/redisson#1893]
Code to reproduce the issue:

```java
RedissonPromise p = new RedissonPromise();
p.cancel(true);
p.cause();
```

**Comment by mrniko** [comment]
Related to https://github.com/redisson/redisson/issues/1879 and https://github.com/redisson/redisson/issues/1567

**Comment by mrniko** [comment]
Fixed

**Comment by hlms** [comment]
Any timeline to add this to 3.10.x version and release?

**Comment by trasa** [comment]
I am seeing a number of exceptions like this when (I think) there is some interruption with access to Redis in 3.10.1:

```
ERROR o.r.client.handler.CommandDecoder - Unable to decode data. reply: :0
, channel: [id: 0x6e4b8748, L:/XXXX - R:XXXXX], command: CommandData [promise=RedissonPromise [promise=ImmediateEventExecutor$ImmediatePromise@3c300c7a(failure: java.util.concurrent.CancellationException)], command=(ZADD), params=[SessionExpired, 1561468690156, PooledUnsafeDirectByteBuf(freed)], codec=org.redisson.codec.JsonJacksonCodec]
java.util.concurrent.CancellationException: null
```

**Comment by mrniko** [comment]
@trasa 

This issue was fixed in 3.10.2 please update to the latest version and give your feedback.
